provider/gke/gke.go

```diff
@@ -411,14 +417,28 @@ func (c *GKE) NewK8sProvider(*kingpin.ParseContext) error {
 	config.AuthInfos[rep.Zone] = authInfo
 	config.CurrentContext = rep.Zone
 
-	k8s, err := k8sProvider.New(c.ctx, *config)
+	k8sClientSet, err := newK8sClient(*config)
 	if err != nil {
 		log.Fatal("k8s provider error", err)
 	}
-	c.k8sProvider = k8s
+	c.k8sProvider = k8sProvider.New(c.ctx, k8sClientSet)
 	return nil
 }
 
+// newK8sClient returns a k8s client from GKE config that can apply and delete resources.
+func newK8sClient(config clientcmdapi.Config) (*kubernetes.Clientset, error) {
```

Comment:
it is a bit confusing that we have NewK8sClient and newK8sClient the same for K8sResourceApply and ResourceApply.

I will think how to simplify.